### PR TITLE
Clarify current state of HTTP/2 support

### PR DESCRIPTION
**Pull request by ptone**

### CODE REVIEW

**Comment by ahmetb** [comment]
can you run `npx markdown-toc -i README.md`  to update ToC?

File: README.md [hunk 1]
```diff
@@ -771,31 +782,19 @@ $ curl --http2 https://<url>
...
-service to multiple regions and adding them behind the load balancer, the
-clients connecting to the load balancer IP/domain will be routed to the Cloud
-Run service **closest** Cloud Run service to the client.
-
 ### Can my application server run on HTTP/2 protocol?
 
-If you develop an app that **only** speaks HTTP/2, Cloud Run can still work with
-that. However, note that Cloud Run requires your application
-to serve on an **unencrypted** endpoint –and HTTP/2 by default requires TLS.
+HTTP/2 to the container is supported for gRPC services.
```
Comment (ahmetb) [inline README.md:787]: I think "currently only supported for gRPC" is a better stmt.

File: README.md [hunk 2]
```diff
@@ -771,31 +782,19 @@ $ curl --http2 https://<url>
...
-fully HTTP/2 without ever being downgraded to HTTP/1. (This is how gRPC works
-on Cloud Run, after all.)
+If your server supports HTTP/2 upgrade via the `h2c` (unencrypted HTTP/2) 
+protocol, it will safely fall-back to HTTP/1.1.
 
-For example in Go, you can use https://godoc.org/golang.org/x/net/http2/h2c
-package to force your HTTP server to serve on `h2c`. (See [this StackOverflow
-answer for
-more](https://stackoverflow.com/questions/61231930/can-i-have-my-cloudrun-server-receive-http-2-requests)).
+If you develop an app that **only** speaks HTTP/2 (with H2C), Cloud Run will
```
Comment (ahmetb) [inline README.md:795]: I think "will not route" should be "will downgrade requests to HTTP/1".

H2C is designed to be compatible with HTTP1.

Comment (ptone) [reply README.md:795]: You can write a HTTP/2 only without downgrade, and require prior-knowledge - but this is in the weeds maybe: ```
// This server only supports "H2C prior knowledge".
// You can add standard HTTP/2 support by adding a TLS config.
func H2CServerPrior() {
	server := http2.Server{}

	l, err := net.Listen("tcp", "0.0.0.0:8080")
	checkErr(err, "while listening")

	fmt.Printf("Listening [0.0.0.0:8080]...\n")
	for {
		conn, err := l.Accept()
		checkErr(err, "during accept")

		server.ServeConn(conn, &http2.ServeConnOpts{
			Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
				fmt.Printf("New Connection: %+v\n", r)
				fmt.Fprintf(w, "Hello, %v, http: %v", r.URL.Path, r.TLS == nil)
			}),
		})
	}
}
```

**Review by ahmetb — APPROVED** [review]

**Pull request merged**